app/Http/Controllers/TalksController.php

```diff
@@ -141,72 +142,70 @@ public function show($id)
 
     public function destroy($id)
     {
-        Auth::user()->talks()->withTrashed()->findOrFail($id)->forceDelete();
+        Auth::user()->talks()->findOrFail($id)->delete();
 
         Session::flash('message', 'Successfully deleted talk.');
 
         return Redirect::to('talks');
     }
 
-    public function softDelete($id)
+    public function archiveIndex()
     {
-        Auth::user()->talks()->findOrFail($id)->delete();
+        $talks =  Auth::user()->talks()->archived()->get();
+        list($talks, $sorting_talk) = $this->sortTalks($talks);
+
+        return View::make('talks.archive')
+          ->with('talks', $talks)
+          ->with('sorting_talk', $sorting_talk);
+    }
+
+    public function archive($id)
+    {
+        $talk = Auth::user()->talks()->findOrFail($id);
+        $talk->is_archived = true;
```

Comment:
Imagine if the syntax is:

```php
Auth::user()->talks()->findOrFail($id)->archive();
```

this would require adding an `archive()` and probably an `unarchive()` method to the model